samples/CallWithChat/src/app/views/CallScreen.tsx

```diff
@@ -141,6 +142,18 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
           console.log(`Call Id: ${callIdRef.current}`);
         }
       });
+
+      /* send add participant request to chat only after call is connected */
+      const joinThreadWhenCallConnected = async (): Promise<void> => {
+        if (adapter.getState().call?.state === 'Connected' && locator && 'chatThreadId' in locator) {
+          adapter.offStateChange(joinThreadWhenCallConnected);
+          console.log('Joining chat thread');
+          joinThread(locator.chatThreadId, userId.communicationUserId, displayName);
+        }
+      };
+
+      adapter.onStateChange(joinThreadWhenCallConnected);
```

Comment:
this is in the sample app code. If a developer is using a composite they would have to perform this functionality. Move this instead to the callwithchatadapter

Reply:
oh this is to request the server to get join access